### PR TITLE
feat: add preferred platform selection and filtering for downloads

### PR DESCRIPTION
**Pull request by Doezer**
This pull request adds support for a "Preferred Platform" setting, allowing users to pre-filter game download search results and auto-search notifications by their chosen platform (e.g., PC, PS5). The system treats PC as a special case, matching both explicit PC releases and those with no detected platform. The feature includes UI updates, backend filtering logic, persistent storage, and comprehensive tests to ensure correct behavior.

**Preferred Platform Filtering and Settings:**

* Added a "Preferred Platform" option in the user settings UI (`settings.tsx`), allowing users to select their preferred platform for downloads and auto-search. The option is persisted and loaded with user settings. [[1]](diffhunk://#diff-ef86b78ab3a2fcb20b1b2b6d4acb9f78241d5b4c4583ea386e82b788f4aa688aR716-R742) [[2]](diffhunk://#diff-ef86b78ab3a2fcb20b1b2b6d4acb9f78241d5b4c4583ea386e82b788f4aa688aR142) [[3]](diffhunk://#diff-ef86b78ab3a2fcb20b1b2b6d4acb9f78241d5b4c4583ea386e82b788f4aa688aR175) [[4]](diffhunk://#diff-ef86b78ab3a2fcb20b1b2b6d4acb9f78241d5b4c4583ea386e82b788f4aa688aR446)
* Migrated the database schema to include a `preferred_platform` field in `user_settings`.
* Updated backend storage and retrieval to handle the new `preferredPlatform` field, ensuring it is saved, updated, and cleared as needed. [[1]](diffhunk://#diff-6af4f8e34285120717169f57249241ec070dfd0f878e186658aaa341e568c84fR862) [[2]](diffhunk://#diff-10721fe466bfe32ad9c1f87fec82c2c8fae26c891936c1e82002c03343127cb3R309-R328)

**Download Dialog and Filtering Logic:**

* Enhanced the `GameDownloadDialog` to preselect and filter results based on the user's preferred platform, with PC matching both explicit and implicit (no platform tag) PC releases. [[1]](diffhunk://#diff-c3061fd23700c28750767d9d851897a6c6801155fe27dd016591ba459a8bcf52R189-R196) [[2]](diffhunk://#diff-c3061fd23700c28750767d9d851897a6c6801155fe27dd016591ba459a8bcf52R263-R270) [[3]](diffhunk://#diff-c3061fd23700c28750767d9d851897a6c6801155fe27dd016591ba459a8bcf52L297-R308) [[4]](diffhunk://#diff-c3061fd23700c28750767d9d851897a6c6801155fe27dd016591ba459a8bcf52L71-R75)
* Updated the available platforms list to include "PC" when any item has no detected platform, ensuring correct filtering and selection.

**Backend Auto-Search Filtering:**

* Implemented strict platform filtering in the auto-search cron job, applying the preferred platform before preferred release group filters. PC preference includes no-platform releases; non-PC preferences are strict. [[1]](diffhunk://#diff-ba75421762c7e2b9cb1f0fa76c848def77f657ddd7b9306cead260d843a10a72R16-R17) [[2]](diffhunk://#diff-ba75421762c7e2b9cb1f0fa76c848def77f657ddd7b9306cead260d843a10a72R107-R122) [[3]](diffhunk://#diff-ba75421762c7e2b9cb1f0fa76c848def77f657ddd7b9306cead260d843a10a72R628) [[4]](diffhunk://#diff-ba75421762c7e2b9cb1f0fa76c848def77f657ddd7b9306cead260d843a10a72L632-R656) [[5]](diffhunk://#diff-ba75421762c7e2b9cb1f0fa76c848def77f657ddd7b9306cead260d843a10a72L733-R760)

**Testing and Validation:**

* Added comprehensive tests for both the frontend dialog and backend cron job to ensure correct filtering behavior for all platform scenarios, including edge cases with no platform preference or ambiguous releases. [[1]](diffhunk://#diff-3ff4db8d9925962f39ef45b432bae5f9a0322639dbe5219aab456bf12daadafdR734-R822) [[2]](diffhunk://#diff-ee9ae74955f435adfd45a530e39debe154a667f609809048605a984b27a810a3R134) [[3]](diffhunk://#diff-ee9ae74955f435adfd45a530e39debe154a667f609809048605a984b27a810a3R661-R812) [[4]](diffhunk://#diff-6df44716e522636ac5e94a64f70521b489ccfae7b710b314b16739686bb011faR9) [[5]](diffhunk://#diff-10721fe466bfe32ad9c1f87fec82c2c8fae26c891936c1e82002c03343127cb3R309-R328)

These changes collectively provide robust support for platform-based filtering throughout the application's download and notification workflows.

### CODE REVIEW

File: migrations/0013_preferred_platform.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `user_settings` ADD `preferred_platform` text;
```
[the remark this answers is not in the document]
Comment (Doezer) [reply migrations/0013_preferred_platform.sql:1]: Fixed in d9f5a63 — added the \

File: shared/title-utils.ts [hunk 2]
```diff
@@ -252,6 +252,30 @@ export function parseReleaseMetadata(releaseName: string): ReleaseMetadata {
...
+ * All non-PC platforms require an explicit detected platform match.
+ */
+export function matchesPlatformFilter(
+  releasePlatform: string | undefined,
+  preferredPlatform: string
+): boolean {
+  if (preferredPlatform === "PC") {
+    return !releasePlatform || releasePlatform === "PC";
+  }
+  return releasePlatform === preferredPlatform;
```
[the remark this answers is not in the document]
Comment (Doezer) [reply shared/title-utils.ts:291]: Fixed in d9f5a63 — \matchesPlatformFilter\ now treats \Xbox\ as a superset: it matches both \Xbox\ and \Xbox Series\ releases. Additionally, \Xbox Series\ has been added to \CANONICAL_PLATFORMS\ so users can also select it explicitly as their preference.

File: client/src/components/GameDownloadDialog.tsx [hunk 3]
```diff
@@ -182,10 +186,14 @@ export default function GameDownloadDialog({ game, open, onOpenChange }: GameDow
         setSelectedGroups(groups);
       }
     }
+    if (userSettings?.preferredPlatform) {
+      setSelectedPlatforms([userSettings.preferredPlatform]);
```
[the remark this answers is not in the document]
Comment (Doezer) [reply client/src/components/GameDownloadDialog.tsx:194]: Already addressed: the useEffect at line 284 has a guard (availablePlatforms.length === 0) so it is a no-op while results are loading. The pre-selection is also gated by a platformPreselectedRef ref so it runs only once.

File: client/src/components/GameDownloadDialog.tsx [hunk 4]
```diff
@@ -182,10 +186,14 @@ export default function GameDownloadDialog({ game, open, onOpenChange }: GameDow
...
+    }
```
[the remark this answers is not in the document]
Comment (Doezer) [reply client/src/components/GameDownloadDialog.tsx]: Already addressed: the platform pre-selection is guarded by platformPreselectedRef.current (set to true after the first application), so subsequent userSettings re-renders do not overwrite a manual selection.

File: server/cron.ts [hunk 5]
```diff
@@ -102,6 +104,22 @@ function applyPreferredGroupsFilter(
...
+function applyPreferredPlatformFilter(
+  items: Awaited<ReturnType<typeof searchAllIndexers>>["items"],
+  preferredPlatform: string | null | undefined
+): Awaited<ReturnType<typeof searchAllIndexers>>["items"] {
+  if (!preferredPlatform) return items;
+  return items.filter((item) => {
+    const { platform } = parseReleaseMetadata(item.title);
+    return matchesPlatformFilter(platform, preferredPlatform);
+  });
+}
```
[the remark this answers is not in the document]
Comment (Doezer) [reply server/cron.ts:121]: Already addressed: metadata is pre-computed once in the itemsMetadata useMemo (which maps each item title to its parsed metadata). The platform filter then reads from this map, avoiding repeated regex operations.

File: shared/title-utils.ts [hunk 6]
```diff
@@ -252,6 +252,30 @@ export function parseReleaseMetadata(releaseName: string): ReleaseMetadata {
   };
 }
 
+/**
+ * Canonical platform labels shared across settings, dialog filtering, and automation.
+ * These are user-facing labels — not raw IGDB platform names.
+ */
+export const CANONICAL_PLATFORMS = ["PC", "PS5", "PS4", "Switch", "Xbox", "Mac", "Linux"] as const;
```
[the remark this answers is not in the document]
Comment (Doezer) [reply shared/title-utils.ts]: Fixed in d9f5a63: CANONICAL_PLATFORMS now includes PS3 and Xbox Series alongside the existing platforms.
